src/garage/experiment/snapshotter.py

```diff
@@ -41,21 +41,46 @@ def __init__(self,
 
     @property
     def snapshot_dir(self, ):
-        """Return the directory of snapshot."""
+        """Return the directory of snapshot.
+
+        Returns:
+            str: The directory of snapshot
+
+        """
         return self._snapshot_dir
 
     @property
     def snapshot_mode(self, ):
-        """Return the type of snapshot."""
+        """Return the type of snapshot.
+
+        Returns:
+            str: The type of snapshot. Can be "all", "last" or "gap"
+
+        """
         return self._snapshot_mode
 
     @property
     def snapshot_gap(self, ):
-        """Return the wait number of snapshot."""
+        """Return the gap number of snapshot.
+
+        Returns:
+            int: The gap number of snapshot.
+
+        """
         return self._snapshot_gap
 
     def save_itr_params(self, itr, params):
-        """Save the parameters if at the right iteration."""
+        """Save the parameters if at the right iteration.
+
+        Args:
+            itr (int): Number of iterations. Used as the index of snapshot.
+            params (obj): Content of snapshot to be saved.
+
+        Raises:
+            ValueError: If snapshot_mode is not one of "all", "last" and "gap".
+            OSError: If fail to remove the old snapshot file.
```

Comment:
for this `OSError` and the `Exception` below, I think it would be simpler if you just defined `class SnapshotterException(Exception): pass` and then raised those.

Reply:
Should I put this exception in `snapshotter.py`, or in a seperate file?

Reply:
you usually put the exception for a file within that file.